data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/GetAggregate.java

```diff
@@ -0,0 +1,146 @@
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.SLOT;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_VALIDATOR;
+import static tech.pegasys.teku.beaconrestapi.SingleQueryParameterUtils.getParameterValueAsBytes32;
+import static tech.pegasys.teku.beaconrestapi.SingleQueryParameterUtils.getParameterValueAsUnsignedLong;
+
+import com.google.common.base.Throwables;
+import com.google.common.primitives.UnsignedLong;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.schema.Attestation;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.beaconrestapi.schema.ErrorResponse;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetAggregate implements Handler {
+
+  public static final String ROUTE = "/validator/aggregate_attestation";
+
+  private final ValidatorDataProvider provider;
+  private final JsonProvider jsonProvider;
+
+  public GetAggregate(final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.provider = provider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Aggregates all attestations matching given attestation data root and slot.",
+      tags = {TAG_VALIDATOR},
+      queryParams = {
+        @OpenApiParam(
+            name = ATTESTATION_DATA_ROOT,
+            description =
+                "`String` HashTreeRoot of AttestationData that validator want's aggregated.",
+            required = true),
+        @OpenApiParam(
+            name = SLOT,
+            description = "`UnsignedLong` Non-finalized slot for which to create the aggregation.",
+            required = true)
+      },
+      description = "Aggregates all attestations matching given attestation data root and slot.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = Attestation.class, isArray = true),
+            description =
+                "Returns aggregated `Attestation` object with same `AttestationData` root."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied"),
+        @OpenApiResponse(
+            status = RES_FORBIDDEN,
+            description = "Beacon node was not assigned to aggregate on that subnet"),
```

Comment:
Yeah, it might be worth raising it on the spec repo. But you are right, atm we aren't enforcing it so it should be fine.
This is likely to change as the spec evolves, so we don't have to worry too much about it.